From: Harald Alvestrand <hta@webrtc.org>
Date: Thu, 31 Aug 2023 09:25:06 +0000
Subject: (cherry-pick-branch-heads/5845) [Merge to 116] CHECK against
 overwrites in send_modules_map_

(cherry picked from commit 9d8fb97b3ca56ec9920271d8e545ae2ac76b143c)

No-try: true
Bug: chromium:1477075
Change-Id: Ia05a868bfab9e99ef66704e8d6bce516a7a43b0a
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/318440
Reviewed-by: Sergey Silkin <ssilkin@webrtc.org>
Commit-Queue: Harald Alvestrand <hta@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#40673}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/319320
Reviewed-by: Philipp Hancke <phancke@microsoft.com>
Reviewed-by: Taylor Brandstetter <deadbeef@webrtc.org>
Cr-Commit-Position: refs/branch-heads/5845@{#7}
Cr-Branched-From: f80cf814353d11a9f22bef5ce5e8868f2c72f0d0-refs/heads/main@{#40319}
---
 modules/pacing/packet_router.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/pacing/packet_router.cc b/modules/pacing/packet_router.cc
index ba133f3d1f..4c986ad027 100644
--- a/modules/pacing/packet_router.cc
+++ b/modules/pacing/packet_router.cc
@@ -68,7 +68,7 @@ void PacketRouter::AddSendRtpModule(RtpRtcpInterface* rtp_module,
 void PacketRouter::AddSendRtpModuleToMap(RtpRtcpInterface* rtp_module,
                                          uint32_t ssrc) {
   RTC_DCHECK_RUN_ON(&thread_checker_);
-  RTC_DCHECK(send_modules_map_.find(ssrc) == send_modules_map_.end());
+  RTC_CHECK(send_modules_map_.find(ssrc) == send_modules_map_.end());
 
   // Signal to module that the pacer thread is attached and can send packets.
   rtp_module->OnPacketSendingThreadSwitched();
-- 
2.34.1

